cachix: CppStdException "Exception: path '/nix/store/xxxx-install' is not valid

Hi!

I'm trying to migrate to cachix v6 and am running into the following error:

```
/bin/sh -c nix path-info --all | grep -v '.drv$' | cat - /tmp/store-path-pre-build | sort | uniq -u  | /nix/var/nix/profiles/per-user/runner/profile/bin/cachix push nix-hie
  cachix: CppStdException "Exception: path '/nix/store/437pzy3m93xdnwknmw89wvhmc9w0vx2h-install' is not valid; type: nix::InvalidPath"
  ##[error]Action failed with error: Error: The process '/bin/sh' failed with exit code 1
  (node:4305) UnhandledPromiseRejectionWarning: Error: The process '/bin/sh' failed with exit code 1
      at ExecState._setResult (/home/runner/work/_actions/cachix/cachix-action/v6/dist/main/index.js:907:25)
      at ExecState.CheckComplete (/home/runner/work/_actions/cachix/cachix-action/v6/dist/main/index.js:890:18)
      at ChildProcess.<anonymous> (/home/runner/work/_actions/cachix/cachix-action/v6/dist/main/index.js:790:27)
      at ChildProcess.emit (events.js:210:5)
      at maybeClose (internal/child_process.js:1021:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
  (node:4305) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
  (node:4305) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
The full logs / set up of the repo are at https://github.com/pbogdan/nix-hie/pull/30

I'm not really sure if it's a user error but as I'm not really familiar with GitHub actions I don't know how to debug this further and / or correctly configure the action on my repo. I would really appreciate any help / pointers on how to fix this. Let me know if I can provide any other info.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -4,45 +4,53 @@ on:
   push:
 
 jobs:
-  tests:
+  public-cache:
     strategy:
       matrix:
         os: [ubuntu-18.04, macos]
     runs-on: ${{ matrix.os }}
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v2
     - run: yarn install --frozen-lockfile
     - run: yarn build
-    - run: yarn test
     - uses: cachix/install-nix-action@v8
     - name: Test public cache
       uses: ./
       with:
         name: cachix-action
-        file: test.nix
         signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
+    - run: nix-build test.nix
+
+  public-cache-no-signing-key:
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, macos]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v2
+    - run: yarn install --frozen-lockfile
+    - run: yarn build
+    - uses: cachix/install-nix-action@v8
     - name: Test public cache no signing key
       uses: ./
       with:
         name: cachix-action
-        file: test.nix
-        signingKey: ''
-    - name: Test public cache no pushing
-      uses: ./
-      with:
-        name: cachix-action
-        skipNixBuild: true
-    - name: Test nixBuildArgs parameter
-      uses: ./
-      with:
-        name: cachix-action
-        file: test-with-arg.nix
-        signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
-        nixBuildArgs: --argstr arg foobar
+    - run: nix-build test.nix
+
+  private-cache:
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, macos]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v2
+    - run: yarn install --frozen-lockfile
+    - run: yarn build
+    - uses: cachix/install-nix-action@v8
     - name: Test private cache
       uses: ./
       with:
         name: cachix-action-private
-        file: test.nix
         signingKey: '${{ secrets.CACHIX_SIGNING_KEY_PRIVATE }}'
         authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+    - run: nix-build test.nix

--- a/src/main.ts
+++ b/src/main.ts
@@ -1,4 +1,5 @@
 import * as core from '@actions/core';
+import { execFileSync } from 'child_process';
 import * as coreCommand from '@actions/core/lib/command'
 import * as exec from '@actions/exec';
 
@@ -41,7 +42,7 @@ async function upload() {
   try {
     if (signingKey !== "" && skipPush !== 'true') {
       core.startGroup('Cachix: pushing paths');
-      await exec.exec("sh", ["-c", `nix path-info --all | grep -v '\.drv$' | cat - /tmp/store-path-pre-build | sort | uniq -u  | ${cachixExecutable} push ${name}`]);
+      execFileSync(`${__dirname}/push-paths.sh`, [cachixExecutable, name], { stdio: 'inherit' });
       core.endGroup();
     }
   } catch (error) {

